### PR TITLE
Candidates Page V2

### PR DESCRIPTION
**Pull request by Theodlz**
* redesign the form based on the Figma designs
* fix the cutout width to show all 3 new/ref/sub on a normal laptop screen, and responsive grid size for smaller-sized
* fix the height of the annotations section
* reorder annotations to first show the annotations of the selected groups for scanning
* trade some of the buttons in the info section for intuitive icons with tooltips.
* stop fetching wavelengths of filters for every single plot from an internal endpoint, and instead use the same colors as the source page (provided by the config endpoint).
* add a show/hide non-detections toggle under the vega plot to easily hide non-detections if needed (can be very helpful to find the first/last detections of an object when we have a lot of upper limits).
* not related, but when requesting annotations from gaia, ps1, ... instead of saving to all user-accessible groups when no group if specified (which is stupid, because all user-accessible groups include single user groups so its a LOT), save only to the public group which anyway the user will have access to. *we should make that change to more endpoints!!!* Anyway, using these buttons everyone will get the same data products, so we might as well just share to the public group!
* drop mui-datatable for a simpler and more customizable MUI List. This helps us make the page mobile friendly too without having to just fight with the existing mui-datatable CSS to get something that barely works. Especially on this candidates page where we don't need to have any concept of columns (because we can't sort by image, photometry, or source info component anyway???), it's pretty useless and added complexity to use a datatable.
* No pages, but instead "infinity scrolling" using the browser's [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), available on all modern browsers (both mobile and desktop). We fetch new candidates when the user gets close to the bottom of the list of what's currently loaded. Candidates are now numbered to avoid confusion and make it easier to keep track of how many still need to be scanned.
* Removed the annotation filtering (not sorting, sorting is still there in the form and on the list), which already did not work at all in production as is. The current code is too complex for the number of annotation keys that exist, and it would end up crashing the browser even on a machine with 12 threads and 64GB of RAM, so I can't even imagine the impact on a "normal" machine.

TODO: cleanup of the css before merging

Here are some screenshots of the new form:
image

and the new table:
image

### CODE REVIEW

File: skyportal/handlers/api/annotation_services.py [hunk 1]
```diff
@@ -176,7 +177,13 @@ def post(self, obj_id):
             group_ids = data.pop('group_ids', None)
 
             if not group_ids:
-                group_ids = [g.id for g in self.current_user.accessible_groups]
+                public_group = session.scalar(
+                    sa.select(Group.id).where(
+                        Group.name == cfg['misc.public_group_name']
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/annotation_services.py:182]: Or sitewide group basically.

**Review by mcoughlin — APPROVED** [review]
Really nice. Just maybe check on default for that public group?

File: skyportal/handlers/api/annotation_services.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Theodlz) [reply skyportal/handlers/api/annotation_services.py:182]: The sitewide group is generated when starting the app using the config, so it should always be there. If we can't find it, it's clearly something that is worth taking the instance down and fixing it before doing anything else. I would rather have the feature not work than posting data to the wrong group (id 0 or 1 could be provisioned admin not necessarily the group, or could be deleted if you changed the public group to be another one at some point).

[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/annotation_services.py:182]: Ok sounds good. Maybe raise a good error for this?

[the remark this answers is not in the document]
Comment (Theodlz) [reply skyportal/handlers/api/annotation_services.py:182]: Sounds good! Will add now

**Comment by Theodlz** [comment]
@mcoughlin just so you know I also made changes to the color palette used on the source page (made the non-ztf green and red filters either lighter or darker (depending on the wavelength) so that one can make the difference.

**Comment by mcoughlin** [comment]
@Theodlz cool!

**Comment by Theodlz** [comment]
We're now removing more code than we're adding, can only be a good sign 😉

**Review by mcoughlin — APPROVED** [review]
@Theodlz I pulled it down and played a bit. So nice.

**Comment by Theodlz** [comment]
> @Theodlz I pulled it down and played a bit. So nice.

Awesome! I'll make a few changes to help with performance as the list grows, cleanup the CSS a bit more, and then it will be ready to go in

**Pull request merged**